core-web/libs/sdk/analytics/src/lib/core/plugin/dot-analytics.plugin.ts

```diff
@@ -38,94 +30,41 @@ export const dotAnalytics = (config: DotCMSAnalyticsConfig) => {
 
         /**
          * Track a page view event
-         * Takes enriched data from properties and creates final structured event
+         * The enricher plugin has already built the complete request body
          */
-        page: (params: DotCMSAnalyticsParams) => {
+        page: (params: DotCMSAnalyticsEnrichedParams) => {
             const { config, payload } = params;
-            const { context, page, device, utm, local_time } = payload;
 
             if (!isInitialized) {
                 throw new Error('DotAnalytics: Plugin not initialized');
             }
 
-            if (!context || !page || !device || !local_time) {
-                throw new Error('DotAnalytics: Missing required payload data for pageview event');
-            }
-
-            // Build final structured event with data property
-            const body: DotCMSPageViewRequestBody = {
-                context,
-                events: [
-                    {
-                        event_type: EVENT_TYPES.PAGEVIEW,
-                        local_time,
-                        data: {
-                            page,
-                            device,
-                            ...(utm && { utm: utm })
-                        }
-                    }
-                ]
+            // Extract only context and events (strip any extra properties from Analytics.js)
+            const body = {
+                context: payload.context,
+                events: payload.events
             };
 
-            if (config.debug) {
-                console.warn('DotAnalytics: Pageview event to send:', body);
-            }
-
             return sendAnalyticsEventToServer(body, config);
         },
 
-        // TODO: Fix this when we haver the final design for the track event
         /**
          * Track a custom event
-         * Takes enriched data and sends it to the analytics server
+         * The enricher plugin has already built the complete request body
```

Comment:
Kind of smell that we have to document that one plugin do one thing... so if we don't use the enricher plugins, the whole thing breaks?

Reply:
We can't use analytics without an enricher.
It is instantiated directly with all the plugins loaded.
```
    const analytics = Analytics({
        app: 'dotAnalytics',
        debug: config.debug,
        plugins: [
            dotAnalyticsIdentityPlugin(config), // Inject identity context (user_id, session_id, local_tz)
            dotAnalyticsEnricherPlugin(), // Enrich and clean payload with page, device, utm data and custom data
            dotAnalytics(config) // Send events to server
        ]
    });
```